### PR TITLE
[OTX] Learning rate bugfix on ReduceLROnPlateau

### PR DESCRIPTION
**Pull request by JihwanEom**
## This PR includes:
- Fix learning rate reverting bug after dropping LR in ROP which was found in detection task.

### What was the root cause?
- ROP didn't consider **adaptive interval**. 
- If current epoch is not multiple of `self.interval` in ROP, `_should_check_stopping` returns `False`.
- Then, by `if` statement at line 505, ROP returns `base_lr` instead of reduced `self.current_lr`.
image

### How I fixed
- Define `self.currnt_lr` in initial epoch and return `self.current_lr` instead of `base_lr` when runner is not in check timing (every interval or epoch).

### Verify ROP
image
- By non-deterministic training, learning rate can differ but note that ROP works well on both tasks by this PR.

### CODE REVIEW

File: otx/algorithms/common/adapters/mmcv/hooks.py [hunk 1]
```diff
@@ -491,8 +491,8 @@ def _init_rule(self, rule, key_indicator):
         self.key_indicator = key_indicator
         self.compare_func = self.rule_map[self.rule]
 
-    def _should_check_stopping(self, runner):
-        """Called _should_check_stopping in ReduceLROnPlateauLrUpdaterHook."""
+    def _is_check_timing(self, runner):
```
Comment (eunwoosh) [inline otx/algorithms/common/adapters/mmcv/hooks.py]: I think this name is a little confusing. how about using `time_to_use` or maybe what you think better?

File: otx/algorithms/common/adapters/mmcv/hooks.py [hunk 2]
```diff
@@ -491,8 +491,8 @@ def _init_rule(self, rule, key_indicator):
...
+        """Called _is_check_timing in ReduceLROnPlateauLrUpdaterHook."""
```
Comment (eunwoosh) [inline otx/algorithms/common/adapters/mmcv/hooks.py]: This is meaningless. it's better to change it more meaningful.

File: otx/algorithms/common/adapters/mmcv/hooks.py [hunk 3]
```diff
@@ -491,8 +491,8 @@ def _init_rule(self, rule, key_indicator):
...
         check_time = self.every_n_epochs if self.by_epoch else self.every_n_iters
         if not check_time(runner, self.interval):
```
Comment (eunwoosh) [inline otx/algorithms/common/adapters/mmcv/hooks.py]: how about change this line to `return check_time(runner, self.interval)` and remove below lines?

**Review by jaegukhyun — DISMISSED** [review]

File: otx/algorithms/common/adapters/mmcv/hooks.py (hunk 1 above)
Comment (sungmanc) [reply otx/algorithms/common/adapters/mmcv/hooks.py]: For the naming, I don't have any good idea. `time_to_use` also looks good to me. However, please put the return type.

```
def _is_check_timing(self, runner) -> bool:
```

**Review by sungmanc — DISMISSED** [review]
Thanks for quick remedy

File: otx/algorithms/common/adapters/mmcv/hooks.py (hunk 1 above)
Comment (harimkang) [reply otx/algorithms/common/adapters/mmcv/hooks.py]: I also recommend adding type hints as well.

Comment (JihwanEom) [reply otx/algorithms/common/adapters/mmcv/hooks.py]: Added type hint, but I love current name :) please go with as-is.

File: otx/algorithms/common/adapters/mmcv/hooks.py (hunk 2 above)
Comment (JihwanEom) [reply otx/algorithms/common/adapters/mmcv/hooks.py]: Reflected. Thank you.

File: otx/algorithms/common/adapters/mmcv/hooks.py (hunk 3 above)
Comment (JihwanEom) [reply otx/algorithms/common/adapters/mmcv/hooks.py]: Reflected. Thank you for good idea!

**Review by eunwoosh — APPROVED** [review]
LGTM. let's check the CI result.

**Review by harimkang — APPROVED** [review]

**Comment by JihwanEom** [comment]
Close by #1533

**Pull request closed without merging**